src/node.rs

```diff
@@ -60,7 +62,7 @@ where
                         unreachable!();
                     };
 
-                    if n.hash == hash && &n.key == key {
+                    if n.hash == hash && n.key.borrow() == key {
```

Comment:
We can probably do this borrow outside of the loop?

Reply:
How so? Isn't `n` changing each time?

Reply:
Ah, yes, of course, you are right. I had `K` and `Q` inverted in my head.